explain-indexes.md

```diff
@@ -0,0 +1,315 @@
+1 row in set (0.00 sec)
+```
+
+## IndexFullScan
+
+Because indexes are ordered, the `IndexFullScan` can be used to optimize common queries such as the `MIN` or `MAX` values for an indexed value:
+
+{{< copyable "sql" >}}
+
+```sql
+EXPLAIN SELECT MIN(intkey) FROM t1;
+EXPLAIN SELECT MAX(intkey) FROM t1;
+```
+
+```sql
++------------------------------+---------+-----------+--------------------------------+-------------------------------------+
+| id                           | estRows | task      | access object                  | operator info                       |
++------------------------------+---------+-----------+--------------------------------+-------------------------------------+
+| StreamAgg_12                 | 1.00    | root      |                                | funcs:min(test.t1.intkey)->Column#4 |
+| └─Limit_16                   | 1.00    | root      |                                | offset:0, count:1                   |
+|   └─IndexReader_29           | 1.00    | root      |                                | index:Limit_28                      |
+|     └─Limit_28               | 1.00    | cop[tikv] |                                | offset:0, count:1                   |
+|       └─IndexFullScan_27     | 1.00    | cop[tikv] | table:t1, index:intkey(intkey) | keep order:true                     |
++------------------------------+---------+-----------+--------------------------------+-------------------------------------+
+5 rows in set (0.00 sec)
+
++------------------------------+---------+-----------+--------------------------------+-------------------------------------+
+| id                           | estRows | task      | access object                  | operator info                       |
++------------------------------+---------+-----------+--------------------------------+-------------------------------------+
+| StreamAgg_12                 | 1.00    | root      |                                | funcs:max(test.t1.intkey)->Column#4 |
+| └─Limit_16                   | 1.00    | root      |                                | offset:0, count:1                   |
+|   └─IndexReader_29           | 1.00    | root      |                                | index:Limit_28                      |
+|     └─Limit_28               | 1.00    | cop[tikv] |                                | offset:0, count:1                   |
+|       └─IndexFullScan_27     | 1.00    | cop[tikv] | table:t1, index:intkey(intkey) | keep order:true, desc               |
++------------------------------+---------+-----------+--------------------------------+-------------------------------------+
+5 rows in set (0.00 sec)
+```
+
+In the above statements, an `IndexFullScan` is performed on each TiKV region. Despite the name `FullScan`, only the first row needs to be read (`└─Limit_28`). Each TiKV region returns its `MIN` or `MAX` value to TiDB, which then performs Stream Aggregation to filter for a single row.
```

Comment:
Stream Aggregation with the aggregation function `MAX` or `MIN` also ensures returning `NULL` if the table is empty.

Reply:
https://github.com/pingcap/docs/pull/3933/commits/0640418ec72b189d3651c340cedf23d5b01ff302